lib/logstash/outputs/s3.rb

```diff
@@ -127,8 +127,19 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   config :canned_acl, :validate => ["private", "public_read", "public_read_write", "authenticated_read"],
          :default => "private"
 
-  # Specifies wether or not to use S3's AES256 server side encryption. Defaults to false.
-  config :server_side_encryption, :validate => :boolean, :default => false
+  # Specifies wether or not to use S3's server side encryption. Defaults to no encryption.
+  config :server_side_encryption, :validate => ["AES256", "aws:kms"]
```

Comment:
That's the tricky part, we have to make this change backward compatible with older configuration.

I suggest we keep the `server_side_encryption` as a boolean that will toggle the encryption or not. The encryption algorithm can be set in another field, and the default could be `AES256`, I think `server_side_encryption_algorithm` would be an appropriate name for it.

it will be a bit more verbose but it will make the change backward compatible, and we try to provide sane default configuration out of the box.

Also I've double checked the ruby AWS-SDK [doc](https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#put_object-instance_method), I believe they accept both AES256 and aes256, so I am fine using the uppercase.
```
server_side_encryption: "AES256", # accepts AES256, aws:kms
```